Assets/Scripts/Shell/BulletController.cs

```diff
@@ -0,0 +1,35 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using TankBattle.Tank;
+
+namespace TankBattle.Bullet
+{
+    public class BulletController
+    {
+        private  BulletView bulletView;
+        private  BulletModel bulletModel;
+        public readonly TankController sourceTank;
+
+        public BulletController(BulletScriptableObject _bulletScriptableObject, Vector3 _position, Quaternion _rotation, TankController _sourceTank)
+        {
+            bulletModel = new BulletModel(_bulletScriptableObject);
+            bulletView = GameObject.Instantiate<BulletView>(bulletModel.BulletPrefab, _position, _rotation);
+            bulletView.SetController(this);
+            bulletView.GetComponent<Rigidbody>().AddForce(bulletView.transform.forward * 1000);
+            sourceTank = _sourceTank;
+        }
+
+        public void DestroyBullet()
+        {
+            bulletModel = null;
+            GameObject.Destroy(bulletView.gameObject);
```

Comment:
same as above let the bullet destroy itself - it might have to do some internal clean up before destroying

Reply:
Sorry, I don't understand what exactly you mentioned. I'm new to c# and don't know how to actually destroy the class objects. I didn't find any delete keywords similar to C++. Can you show me an example of properly destroying and releasing all the class objects?

Reply:
so what I mean is let each class implement its own destroy logic - so in the view if you are say creating a list of game objects for doing something. then even though you called Destroy(bulletview.gameobject) the list was never cleaned up and set to null so it will keep occupying memory if it has some references even when bullet has actually been destroyed. Instead it should look like this - 

bulletView.Destroy();
bulletView = null;

inside bullet view - 
public void Destroy() { 
 Destroy(gameobject);
}